internal/printer/printer.go

```diff
@@ -72,22 +72,17 @@ func (opt *Options) Stderr() *Options {
 }
 
 // New returns an instance of Printer.
-func New(outStream, errStream io.Writer) Printer {
-	if outStream == nil {
-		outStream = os.Stdout
-	}
+func New(errStream io.Writer) Printer {
```

Comment:
@droot Yes. Currently it is only being used by fn commands. All live commands use live printer different from this. I removed it for now as we don't need it and when live also uses this printer we can introduce the stdout option. For now, to be clear, the printer only prints to stderr and this code reflects that.

Reply:
This is not correct. The printer is used by the pkg commands as well as some of the live commands.
Does this lead to all print statements going to stderr? That doesn't seem right.

Reply:
Why not ? The intention is to print the diagnostic information to stderr and actual resources to stdout so that they can be piped to other commands/tools. That makes clear distinction of the output locations.

@frankfarzan @droot

Reply:
The posix guidance for what should go to stdout vs stdin is somewhat vague, but I can see the argument for printing everything to stderr. We should just make sure we do it consistently across all of kpt.

Reply:
I agree that we should have that consistency but not sure if we should do it as part of single PR.

Reply:
So far what we print in `fn` commands is purely progress/diagnostic information, which is sort-of okay to print on `stderr` (POSIX guidelines are not super clear there) and that gives us consistency in `fn` command output in piped/non-piped mode. But for commands such as `kpt live status`, I would expect the output to be printed on `stdout` instead of `stderr`.

I would have liked this printer to have the ability to be able to print on `stderr` and `stdout`, so that this can abstract away printing to CLI from all `kpt` commands. Not having support for `stdout` will make `fmt.Printf` attractive.

Reply:
I agree that for some commands (in particular the live commands), it becomes a bit harder to suggest everything should be printed to stderr. I commented on https://github.com/GoogleContainerTools/kpt/issues/2178, but this seems very focused on the `fn` commands without considering the full kpt UX. Regardless of whether each command is meant to be composed, we should have a consistent approach to what is printed to stdout vs stderr.

Another option would be to have kpt check if the command is being used with pipes and then change what is being printed and whether we use stdout/stderr based on that. The drawback here is that this might be considered confusing.

Reply:
You are right about having consistent approach regardless of the composability.

I found this post very relevant https://mehulkar.com/blog/2017/11/stdout-vs-stderr/

> I’ve always thought that stderr for a program is to display errors and stdout is for “everything else”. But it turns out stdout is meant to be program output and stderr is meant to be… “everything else”. This was conceptually difficult for me until I thought of an example.
> 
> For example, in a calculator program, if I input 1 + 1, the output is 2. This should be printed to stdout. However, if there is internal logging, that should be printed to stderr. E.g. “Gathering input”, “initializing adding machine”, etc.
> 

So we can apply this principle to all the kpt commands. So, for every command, first question to ask is `what is the output` of the command and everything else goes to the `stderr`. WDYT ?

Reply:
I agree with this. This has clear definition.